File: c++/2441.cpp
// Link: https://leetcode.com/problems/largest-positive-integer-that-exists-with-its-negative/description/
// Runtime: 19 ms
// Memory: 26.58 MB

class Solution 
{
public:
    int findMaxK(const std::vector<int>& nums) 
    {
        int maxK = -1;
        std::unordered_set<int> values;
        for (const auto& val : nums)
        {
            if ((abs(val) > maxK) && (values.contains(-val)))
                maxK = abs(val);
            values.insert(val);
        }
    
        return maxK;
    }
};

File: python/2441.py
# Link: https://leetcode.com/problems/largest-positive-integer-that-exists-with-its-negative/description/
# Runtime: 106 ms
# Memory: 16.77 MB


class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        max_k = -1
        values = set()
        for num in nums:
            if (abs(num) > max_k) and (-num in values):
                max_k = abs(num)
            values.add(num)

        return max_k
